snow/engine/snowman/transitive_test.go

```diff
@@ -1254,12 +1254,18 @@ func TestEngineBlockingChitResponse(t *testing.T) {
 	sender.SendPushQueryF = nil
 	sender.CantSendPushQuery = false
 
-	require.NoError(te.Chits(context.Background(), vdr, *queryRequestID, blockingBlk.ID(), issuedBlk.ID(), blockingBlk.ID()))
+	require.NoError(te.Chits(context.Background(), vdr, *queryRequestID, blockingBlk.ID(), gBlk.ID(), blockingBlk.ID()))
 
 	require.Len(te.blocked, 2)
 	sender.CantSendPullQuery = false
+	sender.SendPullQueryF = func(_ context.Context, inVdrs set.Set[ids.NodeID], requestID uint32, blkID ids.ID, requestedHeight uint64) {
+		*queryRequestID = requestID
+		vdrSet := set.Of(vdr)
+		require.Equal(vdrSet, inVdrs)
+		require.Equal(blockingBlk.ID(), blkID)
+		require.Equal(uint64(1), requestedHeight)
+	}
```

Comment:
This test was previously incorrect. Because `GetBlock` doesn't return `missingBlk` (even after issuance), on the old code `blockingBlk` would be dropped. Because the fallback vote was for `issuedBlk`, the preference stayed as `issuedBlk` and the sender didn't error. However, now that `blockingBlk` is issued correctly, we need to handle the repoll for our new preference.